csrc/optimization/alias_analysis.cpp

```diff
@@ -33,10 +33,58 @@ bool isContiguous(const TensorView& tv) {
   return true;
 }
 
-// Finds aliases of `root` and stores the findings in `alias_to_source`.
-void findAliasesOfRoot(
-    const TensorView* root,
-    AliasAnalysisResult& alias_to_source) {
+// Whether a ViewOp transforms any expanded broadcast IterDomain in the input.
+// This is a corner case in which we can't always turn `out` into an alias.
+//
+// For example, given
+//
+//   t0 = makeContigConcreteTensor({4, 5});
+//   t1 = broadcast(t0, {false, false, true});
+//   t2 = expand(t1, {4, 5, 6});
+//
+// `reshape(t2, {40, 3})` and `reshape(t2, {4, 30})` have to copy data.
+// `reshape(t2, {40, 3})` splits the expanded broadcast IterDomain (which is 6)
```

Comment:
nit: our reshape transformation always merges domains until large enough to create the first output domain, so it should look like:

```
merge: 4, 5 -> 20
merge: 20, 6 -> 120
outer-split: 120 -> 40, 3
```

Reply:
Thanks! A PR is coming your way to fix that comment...